sdks/python/apache_beam/transforms/combiners_test.py

```diff
@@ -95,6 +95,27 @@ def test_top(self):
                 label='key:cmp')
     pipeline.run()
 
+  def test_top_key(self):
+    self.assertEqual(
+        ['aa', 'bbb', 'c', 'dddd'] | combine.Top.Of(3, key=len),
+        [['dddd', 'bbb', 'aa']])
+    self.assertEqual(
+        ['aa', 'bbb', 'c', 'dddd'] | combine.Top.Of(3, key=len, reverse=True),
+        [['c', 'aa', 'bbb']])
+
+  def test_sharded_top_combine_fn(self):
+    def test_combine_fn(combine_fn, shards, expected):
+      accumulators = [
+          combine_fn.add_inputs(combine_fn.create_accumulator(), shard)
+          for shard in shards]
+      final_accumulator = combine_fn.merge_accumulators(accumulators)
+      self.assertEqual(combine_fn.extract_output(final_accumulator), expected)
+
+    test_combine_fn(combine.TopCombineFn(3), [range(10), range(10)], [9, 9, 8])
+    test_combine_fn(combine.TopCombineFn(5),
+                    [range(1000), range(100), range(1001)],
+                    [1000, 999, 999, 998, 998])
```

Comment:
Added a test for compare + reverse. key + reverse already tested, compare + key not supported.